capstone-ui/src/Components/InteractiveMap/InteractiveMap.jsx

```diff
@@ -27,7 +84,11 @@ export default function InteractiveMap() {
         .data(states)
         .enter().append('path')
         .attr('class', 'state')
-        .attr('d', path);
+        .attr('d', path)
+        .on('click', function(d) {
+            const stateName = fipsStateCodes[d.id];
+            window.location.href = `/${raceType}/${stateName}`;
```

Comment:
Changing to new page is ok for now. However, you may find issues or complexity trying to pass down values between main map page and state page when building features for filtering, etc. I think some alternatives can be zooming in to state in current map view and or highlighting state with color to show selected state.